ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoContractsSpec.scala

```diff
@@ -4,78 +4,90 @@
+    }
+  }
+
+  it should "not find contracts that are not visible to the requester" in {
+    for {
+      (_, tx) <- store(singleCreate)
+      result <- ledgerDao.lookupActiveOrDivulgedContract(nonTransient(tx).loneElement, charlie)
+    } yield {
+      result shouldEqual None
+    }
+  }
+
+  it should "prevent retrieving the maximum ledger time if some contracts are not found" in {
+    val randomContractId = AbsoluteContractId.assertFromString(s"#random-${UUID.randomUUID}")
+    for {
+      failure <- ledgerDao.lookupMaximumLedgerTime(Set(randomContractId)).failed
+    } yield {
+      failure shouldBe an[IllegalArgumentException]
+      failure.getMessage should startWith(
+        "One or more of the following contract identifiers has been found"
+      )
+    }
+  }
 
-    val transaction = LedgerEntry.Transaction(
-      Some("commandId1"),
-      txId,
-      Some(s"appID-$offsetString"),
-      Some("Alice"),
-      Some(workflowId),
-      let,
-      let,
-      GenTransaction(
-        HashMap(
-          event1 -> NodeCreate(
-            nodeSeed = None,
-            coid = absCid,
-            coinst = someContractInstance,
-            optLocation = None,
-            signatories = Set(alice, bob),
-            stakeholders = Set(alice, bob),
-            key = Some(keyWithMaintainers)
-          )),
-        ImmArray(event1),
-      ),
-      Map(event1 -> Set[Party]("Alice", "Bob"), event2 -> Set[Party]("Alice", "In", "Chains"))
-    )
+  it should "allow the retrieval of the maximum ledger time" in {
     for {
-      result1 <- ledgerDao.lookupActiveOrDivulgedContract(absCid, alice)
-      _ <- ledgerDao.storeLedgerEntry(
-        offset,
-        PersistenceEntry.Transaction(
-          transaction,
-          Map(
-            absCid -> Set(Ref.Party.assertFromString("Alice"), Ref.Party.assertFromString("Bob"))),
-          List.empty
-        )
+      (_, tx) <- store(singleCreate)
+      result <- ledgerDao.lookupMaximumLedgerTime(nonTransient(tx))
+    } yield {
+      inside(result) {
+        case Some(time) => time should be <= Instant.now
+      }
+    }
+  }
+
+  it should "allow the retrieval of the maximum ledger time even when there are divulged contracts" in {
+    val divulgedContractId = AbsoluteContractId.assertFromString(s"#divulged-${UUID.randomUUID}")
+    for {
+      (_, _) <- store(
+        divulgedContracts = Map(
+          (divulgedContractId, someContractInstance) -> Set(charlie)
+        ),
+        offsetAndTx = singleExercise(divulgedContractId)
       )
-      result2 <- ledgerDao.lookupActiveOrDivulgedContract(absCid, alice)
+      (_, tx) <- store(singleCreate)
+      contractIds = nonTransient(tx) + divulgedContractId
+      result <- ledgerDao.lookupMaximumLedgerTime(contractIds)
     } yield {
-      result1 shouldEqual None
-      result2 shouldEqual Some(someContractInstance)
+      inside(result) {
+        case Some(time) => time should be <= Instant.now
```

Comment:
This can probably be compared to the `ledgerEffectiveTime` of `tx`.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5541/commits/759e5cacf0196ae492974383b47f1400667f1ac5